File: app/src/main/java/me/imzack/app/ender/view/contract/AllTypesViewContract.java
package me.imzack.app.ender.view.contract;

import android.support.v7.widget.helper.ItemTouchHelper;
import android.view.View;

import me.imzack.app.ender.view.adapter.TypeListAdapter;

public interface AllTypesViewContract extends BaseViewContract {

    void showInitialView(TypeListAdapter typeListAdapter, ItemTouchHelper itemTouchHelper);

    void onTypeItemClicked(int position, View typeItem);

    void onTypeCreated(int scrollTo);
}

File: app/src/main/kotlin/me/imzack/app/end/view/contract/AllTypesViewContract.kt
package me.imzack.app.end.view.contract

import android.support.v7.widget.helper.ItemTouchHelper
import android.view.View

import me.imzack.app.end.view.adapter.TypeListAdapter

interface AllTypesViewContract : BaseViewContract {

    fun showInitialView(typeListAdapter: TypeListAdapter, itemTouchHelper: ItemTouchHelper)

    fun onTypeItemClicked(position: Int, typeItem: View)

    fun onTypeCreated(scrollTo: Int)
}
